eventstream_rpc/tests/EchoTestRpcClient.cpp

```diff
@@ -28,11 +29,25 @@ namespace Awstest
         return GetAllProductsOperation(
             m_connection, m_echoTestRpcServiceModel.m_getAllProductsOperationContext, m_allocator);
     }
+
+    std::unique_ptr<GetAllProductsOperation> EchoTestRpcClient::NewPtrGetAllProducts() noexcept
+    {
+        return std::unique_ptr<GetAllProductsOperation>(Aws::Crt::New<GetAllProductsOperation>(
+            m_allocator, m_connection, m_echoTestRpcServiceModel.m_getAllProductsOperationContext, m_allocator));
```

Comment:
question: Is there any way the constructor can create an "invalid" version of the Operation, where the user needs to do like:
```
auto operationPtr = client->NewPtrGetAllProducts();
if (!operationPtr) {
    blugh
}
if (!(*operationPtr)) {
    blugh
}
```

If so, I'd strongly prefer we return an empty unique_ptr, instead of returning a unique_ptr that has something in it, but that thing's invalid.

If it's impossible for the custructor to return an invalid thing, then nevermind this is fine

Reply:
It doesn't look like the constructor (in particular, ClientOperation's constructor can enter an invalid state short of allocation failure).